internal/filesystem.go

```diff
@@ -593,3 +593,46 @@ func OpenOrCreateRegularFile(fileName string, flags int, permission fs.FileMode)
 
 	return file, nil
 }
+
+type (
+	readlinkFunc func(name string) (string, error)
+	readdirFunc  func(name string) ([]os.DirEntry, error)
+)
+
+var (
+	defaultReadlink readlinkFunc = os.Readlink
+	defaultReaddir  readdirFunc  = os.ReadDir
+)
+
+// IsProcessRunning returns `true` if the executable specified as an argument is being executed, `false` otherwise.
+func IsProcessRunning(execPath string) bool {
+	isRunning, err := isProcessRunning(execPath, defaultReaddir, defaultReadlink)
+	if err != nil {
+		log.Println(WarningPrefix, "failed to check if process is running, returning false:", err)
+		return false
+	}
+	return isRunning
+}
+
+func isProcessRunning(executablePath string, readdir readdirFunc, readlink readlinkFunc) (bool, error) {
+	procDirs, err := readdir("/proc")
+	if err != nil {
+		log.Printf(ErrorPrefix+" error reading /proc directory: %v\n", err)
```

Comment:
If there is an error the same error message would be printed 2 times, here and in `IsProcessRunning(execPath string) bool`

Reply:
Done :+1: